### ISSUE TITLE
Conflict with std C library macro from complex.h

### ISSUE THREAD

**Issue by apodrugin** [openssl/openssl#7224]
Steps to reproduce:

1. Download file
[main.c.zip](https://github.com/openssl/openssl/files/2383481/main.c.zip)

2. Try to compile it using command
gcc -I YOUR_PATH_TO_OPENSSL_HEADERS main.c

Result:
Compiler error caused by substitution of rsa_mod_exp() function 'I' parameter with 'I' macro from complex.h.

**Comment by levitte** [comment]
I'm quoting main.c here, so a download is no longer needed:

``` C
#include <complex.h>

#include <openssl/rsa.h>

int main(int argc, const char * argv[]) {
    return 0;
}
```

**Comment by levitte** [comment]
Confirmed:

``` C
: ; gcc main.c
In file included from main.c:2:
/usr/include/openssl/rsa.h:459:32: error: expected ‘)’ before ‘__extension__’
     (BIGNUM *r0, const BIGNUM *I, RSA *rsa, BN_CTX *ctx);
                                ^
In file included from main.c:4:
/usr/include/openssl/rsa.h:459:35: error: expected ‘;’, ‘,’ or ‘)’ before ‘RSA’
     (BIGNUM *r0, const BIGNUM *I, RSA *rsa, BN_CTX *ctx);
                                   ^~~
In file included from main.c:2:
/usr/include/openssl/rsa.h:461:68: error: expected ‘)’ before ‘__extension__’
                          int (*mod_exp) (BIGNUM *r0, const BIGNUM *I, RSA *rsa,
                                                                    ^
In file included from main.c:4:
/usr/include/openssl/rsa.h:461:71: error: expected ‘;’, ‘,’ or ‘)’ before ‘RSA’
                          int (*mod_exp) (BIGNUM *r0, const BIGNUM *I, RSA *rsa,
```

The installed openssl on my machinery is:

```
: ; openssl version
OpenSSL 1.1.1-pre9 (beta) 21 Aug 2018
: ; type openssl
openssl är hashad (/usr/bin/openssl)
```

(the team did at some point discussed if the declared parameters should be unnamed (possibly having the names as comments) a while ago...  this issue does point out that we might have to consider it)

**Comment by mspncp** [comment]
> the team did at some point discussed if the declared parameters should be unnamed 

The problem is very special here: The collision is caused by the fact that nobody would expect a parameter name to be in uppercase. IMHO it is sufficient to comment out this particular parameter name and leave all the other parameter names as they are.

**Comment by paulidale** [comment]
Fixed by #7233
